### PR TITLE
feat: support `GUILD_WEB_PAGE_VANITY_URL`

### PR DESCRIPTION
**Pull request by jaw0r3k**
**Please describe the changes this PR makes and why it should be merged:**
PR is different from *original*, as it checks both features to avoid unnecessary request

> Requested by @pkdev08

**Status and versioning classification:**
- Code changes have been tested against the Discord API
-  I know how to update typings and have done so

### CODE REVIEW

File: src/errors/Messages.js [hunk 1]
```diff
@@ -127,7 +127,7 @@ const Messages = {
 
   REACTION_RESOLVE_USER: "Couldn't resolve the user id to remove from the reaction.",
 
-  VANITY_URL: 'This guild does not have the VANITY_URL feature enabled.',
+  VANITY_URL: 'This guild does not have VANITY_URL or GUILD_WEB_PAGE_VANITY_URL features enabled.',
```
Comment (TetieWasTaken) [inline src/errors/Messages.js:130]:
```suggestion
  VANITY_URL: 'This guild does not have the VANITY_URL or GUILD_WEB_PAGE_VANITY_URL feature enabled.',
```

Semantic change

File: src/structures/Guild.js [hunk 2]
```diff
@@ -716,7 +717,7 @@ class Guild extends AnonymousGuild {
    *   .catch(console.error);
    */
   async fetchVanityData() {
-    if (!this.features.includes('VANITY_URL')) {
+    if (!this.features.includes('VANITY_URL') && !this.features.includes('GUILD_WEB_PAGE_VANITY_URL')) {
```
Comment (TetieWasTaken) [inline src/structures/Guild.js:720]: Here you're checking if they have both the `VANITY_URL` _and_ the `GUILD_WEB_PAGE_VANITY_URL` feature. But the error states that the guild does not have the `VANITY_URL` _or_ the `GUILD_WEB_PAGE_VANITY_URL` feature.

Comment (jaw0r3k) [reply src/structures/Guild.js:720]: You are wrong, i am checking if it doesnt have all of these

Comment (TetieWasTaken) [reply src/structures/Guild.js:720]: Oops my bad I didn't see the `!`, you can ignore this suggested change

**Comment by Jiralite** [comment]
This is not documented upstream and I cannot find an existing pull request for this. Until then, this pull request won't be addressed.

**Review by SpaceEEC — APPROVED** [review]

**Comment by Jiralite** [comment]
Until waiting for this to be documented, you could backport #9219?

**Comment by Jiralite** [comment]
Just like https://github.com/discordjs/discord.js/issues/8035#issuecomment-1218424881, there's no point opening a pull request here for a feature that is not documented or is a pending addition. I'll close this because we don't want stale pull requests that can't be resolved.

**Pull request closed without merging**